File: Scripts/laptop/concurrent-computing/c/concurrent-computing.c
#include <stdio.h>
#include <omp.h>

int main()
{

	const volatile char *str[] = { "Enjoy", "Rosetta", "Code" };
	#pragma omp parallel for num_threads(3)
	for ( int j = 0; j < 1000000; ++j) {
		for (int i = 0; i < 3; i++)
			printf("%s\n", str[i]);
	}
	return 0;
}

File: Scripts/laptop/concurrent-computing/php/concurrent-computing.php
#!/usr/bin/php

<?php

class AsyncOperation extends Thread
{
    public function __construct($threadId)
    {
        $this->threadId = $threadId;
    }

    public function run()
    {
        $words = array("Enjoy", "Rosetta", "Code");
        for ($i = 0; $i < 3; $i++) {
        echo "$words[$i]\n";
        sleep(0.001);
        }
    }
}

$start = microtime(true);
for ($i = 0; $i < 1000000; $i++) {
    $t[$i] = new AsyncOperation($i);
    $t[$i]->start();
    $t[$i]->join();
}

?>
